Sources/MapboxCoreNavigation/CacheHandlerFactory.swift

```diff
@@ -0,0 +1,73 @@
+import MapboxDirections
+import MapboxNavigationNative
+
+protocol CacheHandlerData {
+    var tileStorePath: String { get }
+    var credentials: DirectionsCredentials { get }
+    var tilesVersion: String { get }
+    var historyDirectoryURL: URL? { get }
+    var targetVersion: String? { get }
+    var configFactoryType: ConfigFactory.Type { get }
+}
+
+extension NativeHandlersFactory: CacheHandlerData { }
+
+enum CacheHandlerFactory {
```

Comment:
Can you explain (or document) why we need to use the factory pattern here? Could we fold this functionality into an extension of CacheHandle and use file-level variables for things like `cachedHandle`?

Reply:
Added code doc. Ideally, this will be eventually removed when NN implements valid caching the CacheHandler on their side.